js/query.js

```diff
@@ -1,70 +1,452 @@
+                        });
+                    });
+                }
+            )
+            .catch(function(err) {
+                console.error('Fetch Gene Annotation Error', err);
+            });
+    }
+
+    function parseGeneData(data) {
+        return new Promise(function(resolve, reject) {
+            var aliases;
+            var hgnc;
+            var coords;
+            var hg19coords;
+            var mm9coords;
+            var names;
+            var uniprotSum;
+            var wiki;
+            var omim;
+            var ensembl;
+            var vega;
+            var pfam;
+            var uniprot;
+            var pharmgkb;
+            var prosite;
+
+            if (data.hasOwnProperty('HGNC')) {
+                hgnc = {
+                    db: 'https://www.genenames.org/cgi-bin/gene_symbol_report?hgnc_id=',
+                    ident: data.HGNC
+                };
+            } else {
+                hgnc = null;
+            }
+
+            if (data.hasOwnProperty('wikipedia')) {
+                wiki = {
+                    db: 'https://en.wikipedia.org/wiki/',
+                    ident: data.wikipedia['url_stub'].replace(/ /g, '_')
+                };
+            } else {
+                wiki = null;
+            }
+
+            if (data.hasOwnProperty('MIM')) {
+                omim = {
+                    db: 'http://omim.org/entry/',
+                    ident: data.MIM
+                };
+            } else {
+                omim = null;
+            }
+
+            if (data.hasOwnProperty('ensembl')) {
+                ensembl = {
+                    db: 'https://www.ensembl.org/Gene/Summary?g=',
+                    ident: data.ensembl['gene']
+                };
+            } else {
+                ensembl = null;
+            }
+
+            if (data.hasOwnProperty('Vega')) {
+                vega = {
+                    db: 'http://vega.archive.ensembl.org/Gene/Summary?g=',
+                    ident: data.Vega
+                };
+            } else {
+                Vega = null;
+            }
+
+            if (data.hasOwnProperty('pfam')) {
+                pfam = {
+                    db: 'http://pfam.xfam.org/family/',
+                    ident: data.pfam
+                };
+            } else {
+                pfam = null;
+            }
+
+            if (data.hasOwnProperty('pharmgkb')) {
+                pharmgkb = {
+                    db: 'https://www.pharmgkb.org/gene/',
+                    ident: data.pharmgkb
+                };
+            } else {
+                pharmgkb = null;
+            }
+
+            if (data.hasOwnProperty('prosite')) {
+                prosite = {
+                    db: 'https://prosite.expasy.org/',
+                    ident: data.prosite
+                };
+            } else {
+                prosite = null;
+            }
+
+            if (data.hasOwnProperty('alias') && typeof data.alias === 'string') {
```

Comment:
Simplify this to `if (data.hasOwnProperty('alias'))` and remove the `else if` clause.

Check the object key `typeof` using a separate `if` statement within.

What if `data.alias` is neither a string nor an array? Is it guaranteed to be either of those types if it's present?

Reply:
Changed as suggested.

Yes, it will be one of the two.